python/mage/node_classification/models/train_model.py

```diff
@@ -0,0 +1,155 @@
+import torch
+from torch_geometric.loader import NeighborLoader, HGTLoader, ImbalancedSampler
+import mgp
+from tqdm import tqdm
+from torchmetrics import ConfusionMatrix
+
+
+def train_epoch(
+    model: mgp.Any,
+    opt: mgp.Any,
+    data: mgp.Any,
+    criterion: mgp.Any,
+    batch_size: int,
+    observed_attribute: str,
+) -> torch.tensor:
+    """In this function, one epoch of training is performed.
+
+    Args:
+        model (Any): object for model
+        opt (Any): model optimizer
+        data (Data): prepared dataset for training
+        criterion (Any): criterion for loss calculation
+
+    Returns:
+        torch.tensor: loss calculated when training step is performed
+    """
+
+
+    train_input_nodes = (observed_attribute, data[observed_attribute].train_mask)
+    val_input_nodes = (observed_attribute, data[observed_attribute].val_mask)
+
+    train_loader = HGTLoader(
+        data=data,
+        # Sample 512 nodes per type and per iteration for 4 iterations
+        num_samples={key: [512] * 4 for key in data.node_types},
+        shuffle=True,
+        batch_size=batch_size,
+        input_nodes=train_input_nodes,
+    )
+
+    val_loader = HGTLoader(
+        data=data,
+        # Sample 512 nodes per type and per iteration for 4 iterations
+        num_samples={key: [512] * 4 for key in data.node_types},
```

Comment:
Learn about the number 4. Very interesting number, but we should know what it means in this context.